### DISCUSSION TITLE
Any reason why AsyncInitWrapper is not provided by Factory?

### DISCUSSION THREAD

**Discussion by JonnyBeeGod** [hmlongco/Factory#236] [General]
We have a lot of `@MainActor` annotated types we would like to resolve via Factory. When doing so we get compiler errors like "Call to main actor-isolated initializer 'init()' in a synchronous nonisolated context" in the registrations.  Looking at the FactoryDemos you showed how to register those via `AsyncInitWrapper`, however this type is not provided by Factory.  I am wondering what is the reason for this?

**Comment by hmlongco** [comment]
Easiest method, actually, is to simple mark the initializer as nonisolated.

```swift
@MainActor
private final class NonIsolatedMainActorType: Sendable {
    nonisolated init() {}
    func test() {}
}

extension Container {
    fileprivate var nonisolatedMainActor: Factory<NonIsolatedMainActorType> {
        self { NonIsolatedMainActorType() }
    }
}
```
Or you can simply use Task for the job.
```swift
// Factory with Task
extension Container {
    fileprivate var taskMainActor: Factory<Task<SomeMainActorType, Never>> {
        self { Task { await SomeMainActorType() } }
    }
}
```
